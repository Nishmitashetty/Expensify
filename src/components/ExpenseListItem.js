import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import currencyFormatter from 'currency-formatter';



const ExpenseListItem = ({id,description,amount,createdAt}) => (
        <Link className="list-item" to={`/edit/${id}`}>
        <div>
          <h3 className="list-item__title">{description}</h3>
          <span className="list-item__subtitle">{moment(createdAt).format('MMMM Do, YYYY')}</span>
        </div>
        <h3 className="list-item__data"> {currencyFormatter.format(amount/100,{code:'INR'})} </h3>   
        </Link>    
);

export default ExpenseListItem;