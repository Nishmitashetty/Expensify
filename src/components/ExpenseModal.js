import React from 'react';
import Modal from 'react-modal';

const ExpenseModal = (props) =>
{
    return (
        <div>
           <Modal
           isOpen={!! props.selectedExpense}
           contentLabel="Selected Expense"
           closeTimeoutMS={200}
           className="modal"
           >
            <h1 className="modal__title">Delete</h1>
             <p className="modal__message">Are you sure do you want to delete this expense? </p>
             <div className="modal__buttons">
             <button 
             className="cancelButton"
             onClick={props.handleClearSelectedExpense}
             >Cancel
             </button>&nbsp;&nbsp;
             <button
             className="deleteButton"
             onClick={props.onRemove}
             >Delete
             </button>
             </div>
           </Modal>
        </div>
    );
};

export default ExpenseModal;