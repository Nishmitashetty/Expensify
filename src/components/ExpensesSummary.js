import React from 'react';
import {connect} from 'react-redux';
import { Link } from 'react-router-dom';
import currencyFormatter from 'currency-formatter';
import getVisibleExpenses from '../selecters/expenses';
import getExpensesTotal from '../selecters/expenses-total';



export class ExpensesSummary extends React.Component
{
    render()
    {
        const expenseWord = this.props.expensesCount === 1 ? 'expense' : 'expenses';
        const expenseWord2 = this.props.hiddenExpensesCount <= 1 ? 'expense' : 'expenses';
        return (
            <div className="page-header">
                <div className="content-container">
                {
                <h1 className="page-header__title">
                 viewing <span>{this.props.expensesCount}</span> {expenseWord} totalling  <span>{ currencyFormatter.format(this.props.expensesTotal/100,{code:'INR'})}</span> 
                </h1> 
                }
                {
                    this.props.hiddenExpensesCount === 0 ?
                    <h3 className="page-header__subtitle">
                        <span>{this.props.hiddenExpensesCount}</span> {expenseWord2} hidden.
                    </h3>
                    :
                    <h3 className="page-header__subtitle">
                        <span>{this.props.hiddenExpensesCount}</span> {expenseWord2} hidden because of default filters.
                    </h3>              
                }

                <div className="page-header__actions">
                <Link className="button" to="/create">Add Expense</Link>
                </div>
                </div>
            </div>
        );
    }
};

const mapStateToProps = (state) =>
{
    const visibleExpenses = getVisibleExpenses(state.expenses,state.filters);
    return {
        expensesCount:visibleExpenses.length,
        expensesTotal:getExpensesTotal(visibleExpenses),
        hiddenExpensesCount:state.expenses.length - visibleExpenses.length
    }
};

export default connect(mapStateToProps)(ExpensesSummary);