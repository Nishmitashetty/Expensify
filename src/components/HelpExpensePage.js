import React from 'react';


const HelpExpensePage = () => (
    <div>
        This is from HelpExpensePage Component!
    </div>
);

export default HelpExpensePage;