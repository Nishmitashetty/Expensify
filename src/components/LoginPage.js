import React from 'react';
import { startLogin } from '../actions/auth';
import { connect } from 'react-redux';

export class LoginPage extends React.Component
{
    render()
    {
        return (
            <div className="box-layout">
                <div className="box-layout__box">
                <h1 className="box-layout__title">Expensify</h1>
                <p>It's time to get your expenses under control</p>
                <button
                onClick={this.props.startLogin}
                className="button"
                 >Login with Google</button>
                {/* <button
                onClick={this.props.startLogin2} 
                className="button--fb"
                 >Login with Facebook</button> */}
                </div>
            </div>
        );
    };
};

const mapDispatchToProps = (dispatch) =>({
    startLogin:() => dispatch(startLogin()),
   // startLogin2:() => dispatch(startLogin2())
});

export default connect(undefined,mapDispatchToProps)(LoginPage);