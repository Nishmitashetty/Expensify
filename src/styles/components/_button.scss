.button
{
    background:$blue;
    border:none;
    border-radius:3px;
    color:white;
    display:inline-block;
    font-weight:300;
    font-size:$font-size-large;
    line-height:1;
    padding:$s-size;
    text-decoration:none;
}

.button--fb
{
    line-height:1;
    display:inline-block;
    margin-top:$s-size;
    height:41.6px;
    width:168px;
    border-radius:3px;
    background:$blue;
    border:none;
    color:white;
    font-weight:490;
    padding:0;
    margin-bottom:1rem;
}

.button--link
{
    background:none;
}

.button--secondary
{
    background:#888;
}