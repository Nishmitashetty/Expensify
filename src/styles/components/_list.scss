.list-header
{
    background:$off-white;
    border:1px solid darken($off-white,8%);
    color:$grey;
    display:flex;
    justify-content:space-between;
    padding:$s-size $m-size;

}

.list-item
{
    display:flex;
    flex-direction:column;
    padding:$s-size;
    border:1px solid darken($off-white,8%);
    border-top:none;
    text-decoration:none;
    color:$dark-grey;
    transition:background .3s ease;
    &:hover
    {
        background:$off-white;
    }

    @media(min-width:$desktop-breakpoint)
    {
        flex-direction:row;
        justify-content:space-between;
        padding:$m-size;      
     }
}

.list-item__message
{
    justify-content: center;
    align-items:center;
    padding:$m-size;
    color:grey;
    &:hover{
        background:none;
    }
}

.list-item__title
{
    margin:0;
    word-break:break-all;
}

.list-item__subtitle
{
    font-size:$font-size-small;
    color:$grey;
}

.list-item__data
{
    margin:$s-size 0 0 0;

    @media(min-width:$desktop-breakpoint)
    {
        margin:0;
        padding-left:$s-size;
    }
}

.list-body
{
    margin-bottom:$m-size;

    @media(min-width:$desktop-breakpoint)
    {
        margin-bottom:$l-size;
    }
}