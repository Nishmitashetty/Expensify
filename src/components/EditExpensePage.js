import React from 'react';
import {connect} from 'react-redux';
import {
     editExpense,
     startEditExpense,
     startRemoveExpense
    } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';
import ExpenseModal from './ExpenseModal';

// To Avoid inline Functions we have to create class based components

export class EditExpensePage extends React.Component
{
    state ={
        selectedExpense:undefined
    };

    onSubmit= (expense) =>
    {
        this.props.startEditExpense(this.props.expense.id,expense);
        this.props.history.push('/');
    };

    handleClearSelectedExpense = () =>
    {
        this.setState(() => ({
            selectedExpense:undefined
        }));
    };

    onRemove = () =>
    {
        this.props.startRemoveExpense({id:this.props.expense.id});
        this.setState(() => ({
            selectedExpense:undefined
        }));
        this.props.history.push('/');
    };

    onClick = () =>
    {
        this.setState(() =>({
            selectedExpense:this.props.expense.description
        }));
    };

    render()
    {
        return (
            <div>
              <div className="page-header">
                <div className="content-container">
                    <h1 className="page-header__title">Edit Expense</h1>
                </div>
              </div>
              <div className="content-container">
              <ExpenseForm 
              expense={this.props.expense}
              onSubmit={this.onSubmit}
              />
            <button 
            className="button button--secondary"
            onClick={this.onClick}
            >
            Remove Expense
            </button>
            <ExpenseModal 
             selectedExpense={this.state.selectedExpense}
             handleClearSelectedExpense={this.handleClearSelectedExpense}
             onRemove={this.onRemove}
            />
            </div>
            </div>
        );
    }
};

//es6 --short-hand-syntax

const mapStateToProps = (state,props) => ({   expense:state.expenses.find((expense) => expense.id === props.match.params.id) });

const mapDispatchToProps = (dispatch) => ({
     startEditExpense:(id,expense) => dispatch(startEditExpense(id,expense)),
     startRemoveExpense:(data) => dispatch(startRemoveExpense(data))
    });

export default connect(mapStateToProps,mapDispatchToProps)(EditExpensePage);