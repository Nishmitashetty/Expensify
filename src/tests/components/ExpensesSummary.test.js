import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('should render Expenses Summary component correctly with 1 expense',() =>
{
    const result = shallow(<ExpensesSummary expensesCount={1} expensesTotal={1000000} hiddenExpensesCount={0} />);
    expect(result).toMatchSnapshot();
});

test('should render Expenses Summary component correctly with multiple expenses',() =>
{
    const result = shallow(<ExpensesSummary expensesCount={5} expensesTotal={23445000} hiddenExpenses={2} />);
    expect(result).toMatchSnapshot();
});