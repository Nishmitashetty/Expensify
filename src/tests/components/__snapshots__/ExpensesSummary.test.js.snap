// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expenses Summary component correctly with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      viewing 
      <span>
        1
      </span>
       
      expense
       totalling  
      <span>
        ₹10,000.00
      </span>
    </h1>
    <h3
      className="page-header__subtitle"
    >
      <span>
        0
      </span>
       
      expense
       hidden.
    </h3>
    <div
      className="page-header__actions"
    >
      <[object Object]
        className="button"
        to="/create"
      >
        Add Expense
      </[object Object]>
    </div>
  </div>
</div>
`;

exports[`should render Expenses Summary component correctly with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      viewing 
      <span>
        5
      </span>
       
      expenses
       totalling  
      <span>
        ₹234,450.00
      </span>
    </h1>
    <h3
      className="page-header__subtitle"
    >
      <span />
       
      expenses
       hidden because of default filters.
    </h3>
    <div
      className="page-header__actions"
    >
      <[object Object]
        className="button"
        to="/create"
      >
        Add Expense
      </[object Object]>
    </div>
  </div>
</div>
`;
